app/models/spree/preferences/preferable.rb

```diff
@@ -118,7 +118,7 @@ def convert_preference_value(value, type)
         when :boolean
           if value.is_a?(FalseClass) ||
              value.nil? ||
-             value.zero? ||
```

Comment:
very weird that this causes an error :thinking: they are logically equivalent

Reply:
nice, this would work in Java because value.zero? would only be evaluated if the previous line value.nil? was not true. In ruby all parts are evaluated afaik.

Reply:
I believe it works exactly like java but what @mkllnk suggests seems a bit more robust.